.github/workflows/dotnet-upgrade.yml

```diff
@@ -0,0 +1,228 @@
+
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: feature/dotnetsdk-upgrade/${{ needs.dotnet-update.outputs.DOTNET_LATEST_VERSION }}
+
+    # Build runner layout
+    - name: Build & Layout Release
+      run: |
+        ${{ matrix.devScript }} layout Release ${{ matrix.runtime }}
+      working-directory: src
+
+    # Check runtime/externals hash
+    - name: Compute/Compare runtime and externals Hash
+      shell: bash
+      env:
+        DOTNET_RUNTIME_HASH: ${{hashFiles('**/_layout_trims/runtime/**/*')}}
+        EXTERNALS_HASH: ${{hashFiles('**/_layout_trims/externals/**/*')}}
+      run: |
+        echo "Current dotnet runtime hash result: $DOTNET_RUNTIME_HASH"
+        echo "Current Externals hash result: $EXTERNALS_HASH"
+
+        echo "$EXTERNALS_HASH" > ./src/Misc/contentHash/externals/${{ matrix.runtime }}
+        echo "$DOTNET_RUNTIME_HASH" > ./src/Misc/contentHash/dotnetRuntime/${{ matrix.runtime }}
+    - name: GIT commit and push all changed files
+      shell: bash
+      run: |
+        git config --global user.name "AutoUpdater"
+        git config --global user.email "username@users.noreply.github.com"
+
+        git commit -a -m "Update ${{ matrix.runtime }} hashes"
+        git push --set-upstream origin feature/dotnetsdk-upgrade/${{ needs.dotnet-update.outputs.DOTNET_LATEST_VERSION }}
+
+  create-pr:
+    needs: [build-hashes]
+    outputs:
+      # pass outputs from this job to run-tests for use
+      DOTNET_LATEST_VERSION: ${{ needs.build-hashes.outputs.DOTNET_LATEST_VERSION }}
+      DOTNET_MINOR_VERSION: ${{ needs.build-hashes.outputs.DOTNET_MINOR_VERSION }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+      with:
+        ref: feature/dotnetsdk-upgrade/${{ needs.build-hashes.outputs.DOTNET_LATEST_VERSION }}
+    - name: Create Pull Request
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        gh pr create -B main -H feature/dotnetsdk-upgrade/${{ needs.build-hashes.outputs.DOTNET_LATEST_VERSION }} --title "Update dotnet sdk to latest version @${{ needs.build-hashes.outputs.DOTNET_LATEST_VERSION }}" --body "
+          https://dotnetcli.blob.core.windows.net/dotnet/Sdk/${{ needs.build-hashes.outputs.DOTNET_MINOR_VERSION }}/latest.version
+          
+            
+          ---
+            
+          Autogenerated by [DotNet SDK Upgrade Workflow](https://github.com/actions/runner/blob/main/.github/workflows/dotnet-upgrade.yml)"
+
+  run-tests:
+    needs: [create-pr]
+    strategy:
+      # this matrix can still be optimized to remove this, the problem is overlapping `git commits` and needing the latest commits from remote causing merge issues.
+      # Potential solution is to archive all the files needing to be changed and unarchive them and commit them all at once, or compute the hashes from the matrix and run the file changes/git commit at once after they all finish
+      max-parallel: 1
```

Comment:
do you need to limit max-parallel during `run-tests`?

Reply:
Nope, not needed at all. Ill just let the L0's run on the PR itself